### ISSUE TITLE
Google Maps requires a persistent internet connection.

### ISSUE THREAD

**Issue by mcav** [mozilla-sensorweb/sensorweb-client#4]
We need to:
- [ ] Show an "internet connection required" modal when there is no connection when trying to download the Google Maps JS.
- [ ] Properly retry the download.

**Comment by isabelrios** [comment]
@mcav I saw this patch was merged and gave it try. Sorry if there is still pending work but wanted to let you know what I am seeing now and that I am not getting any warning about the connection needed:
With Wi-Fi disabled, once I get to the Where is your sensor? screen, this is what I get:
![screenshot_2016-11-02-11-29-30](https://cloud.githubusercontent.com/assets/1897507/19926181/75886c5a-a0f3-11e6-8fc7-b53e1f834a73.png)

I can type there and address or directly tap on Next and get to the next screen, Which side of the building?
![screenshot_2016-11-02-11-29-41](https://cloud.githubusercontent.com/assets/1897507/19926202/8fd5d142-a0f3-11e6-9953-89754565ef86.png)
Which is not shown fine.

I saw this behaviour with latest master 1faaf5e73 on Android 5.1.1 device.

**Comment by mcav** [comment]
So, this patch only shows the "no internet" dialog if it wasn't able to download Google Maps. We'll need to have that also pop up if there's no connection between the time you start the app (when Google Maps loads its library) and when you view the location selection screen. (#18) 

@isabelrios The problem in your case is that it looks like we couldn't get a location fix, but it still thought it knew your location. I'll have to investigate that (#19)

(Also, that warning only requires _any_ internet connection -- WiFi or Cell, it doesn't matter. The only time we actually probably want WiFi is on the WiFi selection screen, but that may merit more UX discussion.)

**Comment by isabelrios** [comment]
The thing is that I do not get to see the "no internet" dialog at any point, since I launch the app and while I am using it without connection (Yeah, I understand is any connection, data or wifi, was referring to wifi since I do not have a SIM card on the device, so only using wifi :))
If you need me to take logs or give more info for issue #19, please let me know.
Thanks!

**Comment by mcav** [comment]
That's strange. You were able to load Google Maps (because we can see the +/- buttons on the map screen). I don't think it can be cached between app reboots. If you force-quit the app without WiFi and try again, do you get the same result (and could you provide a logcat for that case)?

**Comment by isabelrios** [comment]
I think I know what is happening...I flashed the app from the scratch and launched it without connection for the first time, then you get the correct message: Network Connection Required.

But what happened to me was that I open the app, left it in background and then disabled the connection. Then I was facing the issue I commented. At that point it does not matter if you kill/launch the app, the Network Connection Required is not shown.

The logs in that case are here:
[logs.txt](https://github.com/mozilla-sensorweb/sensorweb-client/files/567115/logs.txt)
